native/src/main/java/io/ballerina/stdlib/protobuf/deserializers/DeserializeHandler.java

```diff
@@ -0,0 +1,286 @@
+            Descriptors.FieldDescriptor> fieldDescriptors, Type type, String messageName) throws IOException {
+
+        BMap<BString, Object> bBMap = null;
+        BArray bArray;
+        Object bMessage = null;
+        if (type != null) {
+            boolean isAnyTypedMessage = "google.protobuf.Any".equals(messageName) &&
+                    fieldDescriptors.values().stream().allMatch(fd -> fd.getFullName().contains("google.protobuf.Any"));
+            boolean isTimestampMessage = (type.getTag() == TypeTags.INTERSECTION_TAG ||
+                    type.getTag() == TypeTags.TUPLE_TAG) && messageName.equals("google.protobuf.Timestamp");
+            if (type.getTag() == TypeTags.RECORD_TYPE_TAG && !isAnyTypedMessage) {
+                bBMap = ValueCreator.createRecordValue(type.getPackage(), type.getName());
+                bMessage = bBMap;
+            } else if (isTimestampMessage) { // for Timestamp
+                TupleType tupleType = TypeCreator.createTupleType(
+                        Arrays.asList(PredefinedTypes.TYPE_INT, PredefinedTypes.TYPE_DECIMAL));
+                bArray = ValueCreator.createTupleValue(tupleType);
+                bMessage = bArray;
+            } else if (type.getTag() == TypeTags.DECIMAL_TAG) { // for Duration type
+                bMessage = ValueCreator.createDecimalValue(new BigDecimal(0));
+            } else if (type.getTag() == TypeTags.MAP_TAG && !isAnyTypedMessage) { // for Struct type
+                bBMap = ValueCreator.createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_ANYDATA));
+                bMessage = bBMap;
+            } else if (type.getTag() == TypeTags.TUPLE_TAG) { // for each field in Struct type
+                TupleType tupleType = TypeCreator.createTupleType(
+                        Arrays.asList(PredefinedTypes.TYPE_STRING, PredefinedTypes.TYPE_ANYDATA));
+                bArray = ValueCreator.createTupleValue(tupleType);
+                bMessage = bArray;
+            } else if (type.getTag() == TypeTags.ARRAY_TAG) { // for array values inside structs
+                bArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_ANYDATA));
+                bMessage = bArray;
+            } else if (isAnyTypedMessage && input != null) {
+                int typeUrlTag = input.readTag();
+
+                if (typeUrlTag == DescriptorProtos.FieldDescriptorProto.Type.TYPE_GROUP_VALUE) {
+                    String typeUrl = input.readStringRequireUtf8();
+                    skipUnnecessaryAnyTypeTags(input);
+                    String s = bytesToHex(codeInputStreamAnyTypeByteArray(input));
+                    bMessage = StringUtils.fromString(s);
+                    BMap<BString, Object> anyMap = ValueCreator.createRecordValue(type.getPackage(), type.getName());
+                    anyMap.put(StringUtils.fromString("typeUrl"), StringUtils.fromString(typeUrl));
```

Comment:
we can use the defined constant

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-protobuf/pull/176/commits/0b4e304a70313de3a78afb5bc4ffe5ce99f8cd6b